### PR TITLE
feat: add helmfile doctor command for AI-assisted diff analysis

### PR DESCRIPTION
**Pull request by yxxhero**
## Summary

`helmfile doctor` runs `helmfile diff` and asks an OpenAI-compatible LLM to summarize the changes and flag risks (data loss, security exposure, breaking changes, downtime, performance, best-practice issues).

**When no LLM is configured, doctor is equivalent to `helmfile diff`** (with `--show-secrets` always forced off). This makes it safe to swap into existing CI jobs.

## Key features

- **OpenAI-compatible**: works with OpenAI, Azure, One-API, LiteLLM, Ollama, vLLM, etc.
- **Secrets always redacted**: two-layer defense (helm-diff `<REDACTED>` + text redactor for Secret blocks, sensitive keys, base64, JWTs)
- **Config precedence**: `env (HELMFILE_LLM_*) < helmfile.yaml (llm:) < CLI flags (--llm-*)`
- **Backend compatibility**: automatic `response_format` fallback for backends that don't support JSON mode
- **Prompt injection defense**: release names JSON-encoded before insertion into LLM prompt
- **Exit codes**: `0` (success/low-risk), `2` (high-risk gate, `--force` to bypass), `1` (errors)

## Usage

```bash
# OpenAI official (no baseURL needed)
export HELMFILE_LLM_API_KEY=sk-...
export HELMFILE_LLM_MODEL=gpt-4o
helmfile doctor

# CI gate
helmfile --environment prod doctor --output json > doctor-report.json
# exit code 2 = high-severity risk detected

# Via helmfile.yaml
llm:
  baseURL: https://one-api.internal/v1
  model: gpt-4o
  apiKey: {{ env "HELMFILE_LLM_API_KEY" }}
```

## New packages

| Package | Purpose |
|---------|---------|
| `pkg/agent/llm` | OpenAI-compatible client, prompt builder, JSON parser, mock client |
| `pkg/agent/doctor` | Secret redactor (state machine + regex), report renderer (markdown + JSON), config resolver |

## Testing

70+ unit tests covering: redaction patterns (Secret blocks, sensitive keys, base64, JWT, ANSI), prompt injection defense, response_format fallback, JSON parsing edge cases (fence stripping, prose extraction), yaml roundtrip, concurrency safety (race detector), panic recovery, error propagation (HTTP 500, timeout, context cancel, malformed JSON).

```
go vet ./...         ✅
golangci-lint run    ✅ 0 issues
go test -race        ✅ all pass
coverage             ~90% on agent packages
```

## Documentation

- Full doctor section in `docs/cli.md`
- `llm:` block reference in `docs/configuration.md`
- Updated `skills/helmfile/SKILL.md` for AI agents

Signed-off-by: yxxhero <aiopsclub@163.com>

### CODE REVIEW

**Review by xiaomudk — DISMISSED** [review]
LGTM overall. I don’t see any blocking issues.
The implementation has a few reasonable tradeoffs, but the safety story and test coverage are both quite good. Nice addition.

File: docs/cli.md [hunk 1]
```diff
@@ -160,6 +160,201 @@ To supply the diff functionality Helmfile needs the [helm-diff](https://github.c
 you should be able to simply execute `helm plugin install https://github.com/databus23/helm-diff`. For more details
 please look at their [documentation](https://github.com/databus23/helm-diff#helm-diff-plugin).
 
+### doctor
+
+`helmfile doctor` runs `helmfile diff` and asks an OpenAI-compatible LLM to summarize the changes and flag risks
+(such as data loss, security exposure, breaking changes, downtime, performance, and best-practice issues).
+
+**When no LLM is configured, `doctor` is equivalent to `helmfile diff` with one
```
Comment (xiaomudk) [inline docs/cli.md]: The fallback behavior docs look a bit stronger than what the implementation actually guarantees.

In the unconfigured path, doctor does delegate to a.Diff(safeCfg), so the overall behavior is close to helmfile diff. However, the docs currently say “same flags, same output, same exit codes”, which does not seem fully accurate.

In particular, the flag surface is not identical:

doctor --output is the doctor report format
helm-diff --output is exposed as --diff-output
Because of that, “same flags” is not strictly true, and “same output” can also be misleading when --output is involved.

I think the docs would be more precise if they said something like:

When no LLM is configured, helmfile doctor falls back to running helmfile diff with --show-secrets forced off. Most diff flags remain available for compatibility, but --output is reserved for doctor report formatting and helm-diff’s output format is exposed as --diff-output.

That would better match the actual behavior and avoid over-promising equivalence.

Comment (yxxhero) [reply docs/cli.md]: Good catch — updated all three locations (cli.md, cmd/doctor.go Long + godoc, pkg/app/doctor.go godoc) to say:

> When no LLM is configured, doctor falls back to helmfile diff with --show-secrets forced off. Most diff flags are accepted; note that --output is reserved for the doctor report format (use --diff-output for helm-diff's plugin output format).

No more "same flags, same output, same exit codes" over-promising. Pushed in a7560a30.

**Review by xiaomudk — APPROVED** [review]
